defaults/pythonSubmitDefault.py

```diff
@@ -498,234 +498,92 @@ def listCommands(self):
-    valuesLists = []
-    for key in keys:
-        valuesLists.append(commandsAndValues[key])
-    valuesLists =np.array(valuesLists).T
-    #edit number of binaries per population
-    for index,command in enumerate(numericalCommands):
-        if command == '--number-of-binaries':
-            break
-    nBinariesPerSimulation = numericalChoices[index]/nSimulations
-    numericalChoices[index] = int(nBinariesPerSimulation)
-    print("index, nBinariesPerSimulation")
-    print(index, nBinariesPerSimulation)
-    bashCommands = []
-    for en in range(nSimulations):
-        bashCommand = ''
-        combination = valuesLists[en]
-        pathName = outPaths[en]
-        for i, val in enumerate(combination):
-            for index,command in enumerate(numericalCommands):
-                if command == keys[i]:
-                    break
-            numericalChoices[index] = val
-        #change the random seed if need be
-        if not shareSeeds:
-            for index,command in enumerate(numericalCommands):
-                if command == '--random-seed':
-                    break
-            numericalChoices[index] += int(nBinariesPerSimulation)
-        #setup output arguments
-        for index,command in enumerate(stringCommands):
-            if command == '--outputPath':
-                break
-        stringChoices[index] = pathName + '/.'
-        bashCommand += generateCommandLineOptions(compas_executable,booleanChoices,booleanCommands,numericalChoices,numericalCommands,stringChoices,stringCommands,listChoices,listCommands)
-        bashCommand += '; '
-        bashCommands.append(bashCommand)
-    return bashCommands
+        for i in range(nList):
+            if listChoices[i]:
+                command.update({listCommands[i] : ' '.join(map(str,listChoices[i]))})
     
+        return command
 
-def runCompas(programOptions):
+
+def combineCommandLineOptionsDictIntoShellCommand(commandOptions):
     """
+    Write out the compas input parameters into a shell string.
+    Ensure the Compas executable is first, and not repeated.
+    Options are non-ordered.
     """
 
-    commands = specifyCommandLineOptions(programOptions)
-
-    for command in commands:
-
-        print(command)
+    shellCommand = commandOptions['compas_executable']
+    del commandOptions['compas_executable'] 
+    for key, val in commandOptions.items():
+        shellCommand += ' ' + key + ' ' + val
 
-        call(command,shell=True)
-
-    return 0
+    return shellCommand
 
 
 if __name__ == "__main__":
 
     #-- Get the program options
     programOptions = pythonProgramOptions()
+    commandOptions = programOptions.generateCommandLineOptionsDict()
+
+    #-- Convert options into a shell string
+    shellCommand = combineCommandLineOptionsDictIntoShellCommand(commandOptions)
 
-    runCompas(programOptions)
+    #-- Run exectute COMPAS shell string
+    print(shellCommand)
+    call(shellCommand,shell=True)
```

Comment:
I assume with this we are still keeping the ability to run pythonSubmit independently ?

Reply:
Yes, the internals are different, but to most users the pythonSubmit can still be configured and run like previously. The only difference is the removal of the hyper-parameter grids, which hopefully will be implementable using the new sw interface.